packages/atomic/src/components/common/generated-answer/generated-answer-common.spec.ts

```diff
@@ -0,0 +1,503 @@
+      fieldsToIncludeInCitations: [],
+      isAnswerGenerated: false,
+      expanded: false,
+      cannotAnswer: false,
+      error: undefined,
+      ...overrides,
+    };
+  };
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  describe('#readGeneratedAnswerStoredData', () => {
+    let mockGetParsedJSON: ReturnType<typeof vi.fn>;
+
+    beforeEach(() => {
+      mockGetParsedJSON = vi.fn();
+      vi.mocked(SafeStorage).mockImplementation(
+        () =>
+          ({
+            getParsedJSON: mockGetParsedJSON,
+          }) as Partial<SafeStorage> as SafeStorage
+      );
+    });
+
+    it('should return stored data when withToggle is true and data exists', () => {
+      const storedData = {isVisible: true};
+      mockGetParsedJSON.mockReturnValue(storedData);
+
+      const result = readGeneratedAnswerStoredData(true);
+
+      expect(mockGetParsedJSON).toHaveBeenCalledWith(
+        StorageItems.GENERATED_ANSWER_DATA,
+        {isVisible: true}
+      );
+      expect(result).toEqual({isVisible: true});
+    });
+
+    it('should return stored data when withToggle is true and data has isVisible false', () => {
+      const storedData = {isVisible: false};
+      mockGetParsedJSON.mockReturnValue(storedData);
+
+      const result = readGeneratedAnswerStoredData(true);
+
+      expect(result).toEqual({isVisible: false});
+    });
+
+    it('should return isVisible true when withToggle is false regardless of stored data', () => {
+      const storedData = {isVisible: false};
+      mockGetParsedJSON.mockReturnValue(storedData);
+
+      const result = readGeneratedAnswerStoredData(false);
+
+      expect(result).toEqual({isVisible: true});
+    });
+
+    it('should use default fallback when no stored data exists', () => {
+      mockGetParsedJSON.mockReturnValue({isVisible: true});
+
+      const result = readGeneratedAnswerStoredData(true);
+
+      expect(mockGetParsedJSON).toHaveBeenCalledWith(
+        StorageItems.GENERATED_ANSWER_DATA,
+        {isVisible: true}
+      );
+      expect(result).toEqual({isVisible: true});
+    });
+  });
+
+  describe('#writeGeneratedAnswerStoredData', () => {
+    let mockSetJSON: ReturnType<typeof vi.fn>;
+
+    beforeEach(() => {
+      mockSetJSON = vi.fn();
+      vi.mocked(SafeStorage).mockImplementation(
+        () =>
+          ({
+            setJSON: mockSetJSON,
+          }) as Partial<SafeStorage> as SafeStorage
+      );
+    });
+
+    it('should store data using SafeStorage', () => {
+      const data = {isVisible: false};
+
+      writeGeneratedAnswerStoredData(data);
+
+      expect(mockSetJSON).toHaveBeenCalledWith(
+        StorageItems.GENERATED_ANSWER_DATA,
+        data
+      );
+    });
+
+    it('should handle isVisible true data', () => {
+      const data = {isVisible: true};
+
+      writeGeneratedAnswerStoredData(data);
+
+      expect(mockSetJSON).toHaveBeenCalledWith(
+        StorageItems.GENERATED_ANSWER_DATA,
+        data
+      );
+    });
```

Comment:
These two tests are the same?

I would expect at least one test that checks that if you call write, and then after call read you get the right value back.